### PR TITLE
[WIP] Pad to multiple of 8 for run mlm example

### LINKED ISSUE #10627: considering `pad_to_multiple_of` for run_mlm.py
## Environment info

- `transformers` version: 4.3.3
- Platform: linux
- Python version: 3.7
- PyTorch version (GPU?): 1.8
- Tensorflow version (GPU?): - 
- Using GPU in script?: - 
- Using distributed or parallel set-up in script?: - 

### Who can help
@sgugger, @patil-suraj

## Information
I have seen in huggingface codes such as run_seq2seq.py that for the case of fp16, they pad to multiple of 8 like below, perhpas for efficiency purpose:

```
 data_collator = DataCollatorForSeq2Seq(
        tokenizer,
        label_pad_token_id=label_pad_token_id,
        pad_to_multiple_of=8 if training_args.fp16 else None,
       )
```

For run_mlm.py codes, the padding condition in case of fp16 is not considered, could it be a bug and possibly performance could get better in run_mlm.py if this was set? kindly appreciate  having a look.

thanks

### PR DESCRIPTION
**Pull request by ak314**
# What does this PR do?
This PR uses padding to a multiple of 8 in the run_mlm.py language modeling example, when fp16 is used. Since the DataCollatorForLanguageModeling did not initially accept the pad_to_multiple_of option, that functionality was added.
Fixes #10627

## Before submitting
- [x] Did you write any new necessary tests?

## Who can review?
@sgugger

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Very clean! Thanks a lot for this PR and adding those new tests!
I see the title is still [WIP}, could you remove that once you're done?

**Comment by ak314** [comment]
@sgugger I fixed an edge case in the `_collate_batch` method and added tests for that, but if it's ok I'd like to close this PR and open a fresh one.

**Comment by sgugger** [comment]
Closing in favor of #11128
